### PR TITLE
LUD-135 Add kickstart support for BOSS and SATADOM

### PR DESCRIPTION
**Pull request by aimers1975**
Add kickstart support for BOSS and SATADOM local flash boot option. Also fixed hostname warning
during install.
Should be merged with PR:
https://github.com/dell-asm/dell-idrac/pull/278

### CODE REVIEW

File: tasks/vmware_esxi.task/ks.cfg.erb [hunk 1]
```diff
@@ -97,11 +97,13 @@ for disk in $disk_paths; do
...
   device=`basename $disk`
-  local_ata=`localcli storage core device list -d $device | grep -iE "Local ATA Disk|DELL Serial|Local DELL Disk"`
+  local_ata=`localcli storage core device list -d $device | grep -iE "Local ATA Disk|DELL Serial|Local DELL Disk|DELLBOSS|SATADOM"`
   if [ $? -eq 0 ]; then
     local_sas=`localcli storage core device list -d $device | grep -i "Is SAS: false"`
     local_ata_dell=`localcli storage core device list -d $device | grep -iE "DELL Serial|Local DELL Disk"`
-    if [ -z "$local_sas" -a -z "$local_ata_dell" ] ; then
+    boss=`localcli storage core device list -d $device | grep -iE "DELLBOSS"`
+    satadom=`localcli storage core device list -d $device | grep -iE "SATADOM"`
+    if [ -z "$local_sas" -a -z "$local_ata_dell" -a -z "$boss" -a -z "$satadom" ] ; then
```
Comment (gavin-scott) [inline tasks/vmware_esxi.task/ks.cfg.erb]: This was the pattern in the previous code, but I can't see any of the checks from lines 102 - 106.  If line 100 succeeded (which id did based on check on line 101) why don't we always just use that?

File: tasks/vmware_esxi.task/ks.cfg.erb [hunk 2]
```diff
@@ -64,8 +64,8 @@ if !options.nil? && !options.empty?
   else
     line += ' --nodns'
   end
-  if options['os_host_name']
-    line += " --hostname='#{options['os_host_name']}'"
+  if options['hostname']
+    line += " --hostname='#{options['hostname']}'"
```
[the remark this answers is not in the document]
Comment (aimers1975) [reply tasks/vmware_esxi.task/ks.cfg.erb]: @gavin-scott I'm not sure the origin of the os_host_name...but I was noticing warnings during the install complaining something about hostname...and when I checked the razor metadata...it looked like their was a hostname field but not an os_host_name field...

[the remark this answers is not in the document]
Comment (gavin-scott) [reply tasks/vmware_esxi.task/ks.cfg.erb]: You're right, looks like a bug the gets papered over later on so we never noticed it. Thanks!

File: tasks/vmware_esxi.task/ks.cfg.erb (hunk 1 above)
Comment (gavin-scott) [reply tasks/vmware_esxi.task/ks.cfg.erb]: We talked. All this logic is pretty confusing but we landed on baking the boss/satdom check into the local_ata_dell check.

**Comment by aimers1975** [comment]
@gavin-scott OK, updated!

**Pull request merged**